core/domain/classifier_services_test.py

```diff
@@ -0,0 +1,166 @@
+# coding: utf-8
+#
+# Copyright 2015 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from core.domain import classifier_services
+
+import test_utils
+import utils
+
+import numpy
+
+
+class StringClassifierUnitTests(test_utils.GenericTestBase):
+
+    _EXAMPLES = [
+        ['i eat fish and vegetables', ['food']],
+        ['fish are pets', ['pets']],
+        ['my kitten eats fish', ['food', 'pets']]
+    ]
+
+    _NEW_EXAMPLES = [
+        ['i only eat fish and vegetables', []],
```

Comment:
Do we handle the case where new examples may come with a nonempty set of labels? If so, we should test this; if not, we should guard against it.